rust/kernel/rbtree.rs

```diff
@@ -301,6 +313,427 @@ impl<K, V> RBTree<K, V> {
         None
     }
 
+    fn neighborhood_search_raw(
```

Comment:
I realize this helper might appear overcomplicated...as I've gone beyond implementing the minimal things we need for binder...but couldn't resist it for completeness.  With 2 directions and 3 "flavors" of methods as called out in the description, there was a lot of duplication happening without such a helper.